akka-stream/src/main/scala/akka/stream/javadsl/Framing.scala

```diff
@@ -94,6 +94,33 @@ object Framing {
     scaladsl.Framing.lengthField(fieldLength, fieldOffset, maximumFrameLength, byteOrder).asJava
 
   /**
+   * Creates a Flow that decodes an incoming stream of unstructured byte chunks into a stream of frames, assuming that
+   * incoming frames have a field that encodes their length.
+   *
+   * If the input stream finishes before the last frame has been fully decoded this Flow will fail the stream reporting
+   * a truncated frame.
+   *
+   * @param fieldLength The length of the "size" field in bytes
+   * @param fieldOffset The offset of the field from the beginning of the frame in bytes
+   * @param maximumFrameLength The maximum length of allowed frames while decoding. If the maximum length is exceeded
+   *                           this Flow will fail the stream. This length *includes* the header (i.e the offset and
+   *                           the length of the size field)
+   * @param byteOrder The ''ByteOrder'' to be used when decoding the field
+   * @param computeFrameSize This function can be supplied if frame size is varied or needs to be computed in special fashion.
+   *                         For example, frame can have shape like this: `[offset bytes][body size bytes][body bytes][footer bytes]`.
+   *                         Then computeFrameSize can be used to compute frame size: `(offset bytes, computed size) => (actual frame size)`.
+   *                         ''Actual frame size'' must be equal or bigger than sum of `fieldOffset` and `fieldLength`.
```

Comment:
`in _a_ special fashion`, "have _a_ shape like this", "compute _the_ frame size".

"actual frame size must be equal ...", please clarify what happens if not, does the stage fail?